lib/mutgen.c

```diff
@@ -1201,7 +1219,8 @@ mutgen_choose_alleles(mutgen_t *self, tsk_id_t *parent, mutation_t **bottom_muta
         } else {
             parent_mut = bottom_mutation[u];
             mut->parent = parent_mut;
-            if (mut->time > parent_mut->time || (parent_mut->new && !mut->new)) {
+            if (!allow_ancestral
+                && (mut->time > parent_mut->time || (parent_mut->new && !mut->new))) {
```

Comment:
move `mut->time > parent_mut->time` to an assert. Or delete it since we'll find out about the error later - we won't have a valid tree sequence.

Reply:
oh delete this whole thing since we can't trigger it any more

Reply:
it is being triggered by one of the tests, check which.